### ISSUE TITLE
Question: Has anyone managed to use this library with OSGi/bndtools?

### ISSUE THREAD

**Issue by jln-ho** [influxdb/influxdb-java#134]
TL;DR: Is anyone using influxdb-java in an OSGI/bndtools environment that can explain how to get it to work in an OSGi bundle? Especially with respect to the ``.bnd`` file? 

I want to use influxdb-java in an OSGi project and I'm using bndtools (not Maven) for my builds. I created a fat jar including all the transitive dependencies using the maven-shade-plugin which seems to work fine if I include it on the classpath of "standard" (i.e. non-OSGi) Java projects. 

When I put the fat jar on the classpath of an OSGi bundle, however, dependency hell breaks loose and I need to go through lots of trickery in my ``.bnd`` file just to get the bundle to resolve. Once resolved, I can at create databases and run queries, but when I try to write datapoints the ``write()`` method never returns. 

I'm quite sure this is because my ``.bnd`` file is misconfigured, so here it is just for reference: 

````
-buildpath: \
	osgi.core,\
	osgi.cmpn;version=5.0.0,\
	org.apache.felix.dependencymanager.annotation,\
	lib/uber-influxdb-java-2.0-SNAPSHOT.jar;version=file,\
	javax.servlet
	
	
Bundle-Version: 0.0.0.${tstamp}
Export-Package: example.influxdb2

Import-Package: retrofit.android; resolution:=optional,*\
				retrofit.appengine; resolution:=optional,*\
				javax.servlet; resolution:=optional,*\
				android.os; resolution:=optional,*\
				aQute.*; resolution:=optional,*\
				com.sun.jdmk.*; resolution:=optional,*\
				javax.jms; resolution:=optional,*\
				javax.mail.*; resolution:=optional,*\
				sun.misc; resolution:=optional,*
				
Private-Package: org.influxdb,\
					org.influxdb.*,\
					com.squareup.*,\
					com.google.common.base,\
					com.google.common.collect,\
					com.google.common.io,\
					com.google.common.hash,\
					com.google.common.math,\
					com.google.common.primitives,\
					com.google.common.*,\
					com.google.gson.*,\
					com.google.gson,\
					com.google.thirdparty.*,\
					okio,\
					retrofit,\
					retrofit.client,\
					retrofit.converter,\
					retrofit.http,\
					retrofit.mime,\
					org.apache.http,\
					org.apache.http.*,\
					org.apache.commons.*,\
					org.apache.avalon.framework.logger,\
					org.apache.avalon.framework,\
					org.apache.log,\
					org.apache.*,\
					org.json,\
					rx.*,\
					
````
Also, [here](http://minitux.mooo.com/uber-influxdb-java-2.0-SNAPSHOT.jar.zipinfo.txt) is what's included in my fat jar.

Any help would be greatly appreciated!

**Comment by majst01** [comment]
Can you please try to enable logging of the driver with:

influxdb.setLoglevel(LogLevel.FULL)

before any write occurs and show the output created.

Greetings
Stefan

**Comment by jln-ho** [comment]
Well, here is the log output for the two queries I run before writing. Seems like there isn't any output for the attempted write, though

````
---> HTTP GET http://localhost:8086/query?u=osgi-test&p=osgi-test&q=SHOW+DATABASES
---> END HTTP (no body)
<--- HTTP 200 http://localhost:8086/query?u=osgi-test&p=osgi-test&q=SHOW+DATABASES (2ms)
Content-Type: application/json
Request-Id: 52335a9f-a008-11e5-8014-000000000000
X-Influxdb-Version: 0.9.4.2
Date: Fri, 11 Dec 2015 13:08:50 GMT
OkHttp-Selected-Protocol: http/1.1
OkHttp-Sent-Millis: 1449839330394
OkHttp-Received-Millis: 1449839330395

{"results":[{"series":[{"name":"databases","columns":["name"],"values":[["_internal"],["oci-trainer"],["unittest_1449075074930"],["unittest_1449224566926"],["unittest_1449224618914"],["unittest_1449224715696"],["unittest_1449225143441"],["unittest_1449225173057"],["unittest_1449225424292"],["unittest_1449225592442"],["unittest_1449226936926"],["unittest_1449232527197"],["unittest_1449232577201"],["unittest_1449232635244"],["unittest_1449232655025"],["unittest_1449232731943"],["unittest_1449232852805"],["unittest_1449234157179"],["unittest_1449234535463"],["unittest_1449236453497"],["unittest_1449238250856"],["unittest_1449238468727"],["unittest_1449758798033"],["helloinflux"],["helloinflux2"]]}]}]}
<--- END HTTP (707-byte body)
---> HTTP GET http://localhost:8086/query?u=osgi-test&p=osgi-test&db=helloinflux2&q=show+measurements
---> END HTTP (no body)
<--- HTTP 200 http://localhost:8086/query?u=osgi-test&p=osgi-test&db=helloinflux2&q=show+measurements (1ms)
Content-Type: application/json
Request-Id: 5233df18-a008-11e5-8015-000000000000
X-Influxdb-Version: 0.9.4.2
Date: Fri, 11 Dec 2015 13:08:50 GMT
OkHttp-Selected-Protocol: http/1.1
OkHttp-Sent-Millis: 1449839330398
OkHttp-Received-Millis: 1449839330399

{"results":[{}]}
<--- END HTTP (16-byte body)
QueryResult [results=[Result [series=null, error=null]], error=null]
````

Here's the code snippet I'm running. The println statement at the very end is never reached.
````java
String dbName = "helloinflux2";
InfluxDB influxDB = InfluxDBFactory.connect("http://localhost:8086",
				"osgi-test", "osgi-test");
influxDB.setLogLevel(LogLevel.FULL);
influxDB.describeDatabases();
influxDB.query(new Query("show measurements", dbName));
		
BatchPoints batchPoints = BatchPoints.database(dbName)
		.tag("async", "true").retentionPolicy("default")
		.consistency(ConsistencyLevel.ALL).build();
for (int i = 0; i < 10; i++) {
	Point p = Point.measurement("osgitest")
			.time(System.currentTimeMillis(), TimeUnit.MILLISECONDS)
			.field("idle", Math.random())
			.field("system", Math.random())
			.build();
	batchPoints.point(p);
	try {Thread.sleep(1);} catch (Exception  e) {}
}

influxDB.write(batchPoints);
System.out.println("done");
````

**Comment by majst01** [comment]
Hi,

you are using influxdb-java in version 2.0-SNAPSHOT, please try first 2.1 which was release to maven repo yesterday. Otherwise im not sure what version you are using exactly, there have been many fixes since a 2.0-SNAPSHOT version.

**Comment by jln-ho** [comment]
The behavior is the same with the 2.1 release, tested it two hours ago.

**Comment by majst01** [comment]
OK, try the same outside a OSGI Runtime, put it in a simple Java Main, or a Unittest and try to reproduce.

**Comment by jln-ho** [comment]
I'm quite sure there's no issue with the client itself because I can run the same code with the same fat jar from a non-OSGi project just fine (as stated in my initial comment).

**Comment by majst01** [comment]
My knowledge of OSGI is very limited, im afraid but i cannot help
